vyper/compiler/phases.py

```diff
@@ -225,10 +237,13 @@ def generate_lll_nodes(global_ctx: GlobalContext) -> Tuple[parser.LLLnode, parse
     lll_nodes, lll_runtime = parser.parse_tree_to_lll(global_ctx)
     lll_nodes = optimizer.optimize(lll_nodes)
     lll_runtime = optimizer.optimize(lll_runtime)
+    if use_ovm:
+        lll_nodes = ovm.rewrite_lll_for_ovm(lll_nodes)
+        lll_runtime = ovm.rewrite_lll_for_ovm(lll_runtime)
     return lll_nodes, lll_runtime
 
 
-def generate_assembly(lll_nodes: parser.LLLnode) -> list:
+def generate_assembly(lll_nodes: parser.LLLnode, use_ovm: bool = False) -> list:
```

Comment:
Just a comment, no change needed.  Could be a bit nitpicky or YAGNI syndrome, the approach of passing a bool flag across the call stack works here for what we want. 

Wondering if other EVM like zkEVM, etc. May need rewrite of opcodes the solution will become adding more bool flags? which at that point it may be better to go with a Config struct arg or dictionary or change to approach to functional composition `generate_assembly_ovm`, etc.

Reply:
Agree on minimizing the diff to get this out. Sounds good to refactor later if needed.